.github/workflows/test.yaml

```diff
@@ -105,3 +114,36 @@ jobs:
       run: |
         python --version
         tox -q
+
+
+  pypi-publish:
+    # https://github.community/t/is-it-possible-to-require-all-github-actions-tasks-to-pass-without-enumerating-them/117957/4?u=graingert
+    runs-on: ubuntu-20.04
+    needs: [testing]
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: Set up Python
+      uses: actions/setup-python@v2
+      with:
+        python-version: 3.9
+
+    - name: Test
+      run: |
+        python --version
+        rm -rf dist/*
+        tox -e release-prepare
+
+    - name: Display structure of files to be pusehed
+      run: ls -R dist/
+
+    - name: Publish to PyPI on a new tag
+      # Testing:
+      # TEST_PYPI_UPLOAD_TOKEN
+      # repository_url: https://test.pypi.org/legacy/
+      # Prod secret: PYPI_UPLOAD_TOKEN
+      if: startsWith(github.ref, 'refs/tags')
+      uses: pypa/gh-action-pypi-publish@v1.3.1
+      with:
+        password: ${{ secrets.TEST_PYPI_UPLOAD_TOKEN }}
+        repository_url: https://test.pypi.org/legacy/
```

Comment:
```suggestion
      if: startsWith(github.ref, 'refs/tags')
      uses: pypa/gh-action-pypi-publish@master
      with:
        password: ${{ secrets.PYPI_UPLOAD_TOKEN }}
```

:shipit:

Reply:
I kept those comments to make it easy to switch to testing in a future branch.